services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -174,6 +175,11 @@ public TokenValidation validateToken(KeycloakSession session, UriInfo uriInfo, C
             }
         }
 
+        if (oldToken.getIssuedAt() + 1 < clientSession.getStarted()) {
```

Comment:
This condition seems duplicated at different places. Does it make sense to overload `AccessToken.isActive` or another method to keep this logic in a single place?

Reply:
Good point. I will try to extract possibly to some util method or something... Not sure about `AccessToken.isActive` as AuthenticatedClientSession can't be sent as an argument to AccessToken method (it is different module) .

Reply:
Sure. We only need the int anyways.

Reply:
True. I've added method "isIssuedBeforeSessionStart" with the time as an argument. Also used this for few other places inside the codebase as there are similar existing checks for the userSession already existing.